### PR TITLE
Mobile 2.0 Client apis

### PR DESCRIPTION
**Pull request by hrajput89**
#### Fixes #.

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- Client APIs

### CODE REVIEW

File: CBLClient/Client.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            # Create connection to method endpoint.
+            headers = {"Content-Type": "application/json"}
+            self.session.headers = headers
+            resp = self.session.post(url, data=json.dumps(body))
+            resp.raise_for_status()
+            responseCode = resp.status_code
+
+            if responseCode == 200:
+                result = resp.content
+                # log_info("Got response: {}".format(result))
```
Comment (raghusarangapani) [inline CBLClient/Client.py]: Remove this comment

File: CBLClient/Database.py [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+            args.setString("concurrencyControlType", concurrencyControlType)
+        return self._client.invokeMethod("database_deleteWithConcurrency", args)
+
+
+#     Not implemented on server
+#     def create_value_index(self, database, prop):
+#         args = Args()
+#         args.setMemoryPointer("database", database)
+#         args.setString("property", prop)
+#         return self._client.invokeMethod("create_valueIndex", args)
```
Comment (raghusarangapani) [inline CBLClient/Database.py]: Can these commented lines be removed?

File: CBLClient/DatabaseConfiguration.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    def __init__(self, base_url):
+        self.base_url = base_url
+
+        # If no base url was specified, raise an exception
+        if not self.base_url:
+            raise Exception("No base_url specified")
+
+        self._client = Client(base_url)
+    """
```
Comment (raghusarangapani) [inline CBLClient/DatabaseConfiguration.py]: Can this commented method be removed?

File: CBLClient/Dictionary.py [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+        self._client = Client(base_url)
+
+    def create(self, dictionary=None):
+        args = Args()
+        if dictionary:
+            args.setMemoryPointer("content_dict", dictionary)
+        return self._client.invokeMethod("dictionary_create", args)
+
+#     Method no longer exist
```
Comment (raghusarangapani) [inline CBLClient/Dictionary.py]: Can these commented lines be removed?

File: CBLClient/Dictionary.py [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+        return self._client.invokeMethod("dictionary_getString", args)
+
+    def setString(self, dictionary, key, value):
+        args = Args()
+        args.setMemoryPointer("dictionary", dictionary)
+        args.setString("key", key)
+        args.setString("value", value)
+        return self._client.invokeMethod("dictionary_setString", args)
+
+#     Method no longer exist
```
Comment (raghusarangapani) [inline CBLClient/Dictionary.py]: Can these commented lines be removed?

File: CBLClient/Dictionary.py [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+        return self._client.invokeMethod("dictionary_getLong", args)
+
+    def setLong(self, dictionary, key, value):
+        args = Args()
+        args.setMemoryPointer("dictionary", dictionary)
+        args.setString("key", key)
+        args.setLong("value", value)
+        return self._client.invokeMethod("dictionary_setLong", args)
+
+#     Method no longer exist
```
Comment (raghusarangapani) [inline CBLClient/Dictionary.py]: Can these commented lines be removed?

File: CBLClient/Document.py [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+            return self._client.invokeMethod("document_create", args)
+        elif dictionary:
+            args.setMemoryPointer("dictionary", dictionary)
+            return self._client.invokeMethod("document_create", args)
+        elif doc_id:
+            args.setString("id", doc_id)
+            return self._client.invokeMethod("document_create", args)
+        return self._client.invokeMethod("document_create", args)
+
+    # Why this method is in Document class
```
Comment (raghusarangapani) [inline CBLClient/Document.py]: This comment doesn't make any sense

File: CBLClient/Document.py [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+        return self._client.invokeMethod("document_getString", args)
+
+    def setString(self, document, key, value):
+        args = Args()
+        args.setMemoryPointer("document", document)
+        args.setString("key", key)
+        args.setString("value", value)
+        return self._client.invokeMethod("document_setString", args)
+
+#     def set(self, document, dictionary):
```
Comment (raghusarangapani) [inline CBLClient/Document.py]: Can these commented lines be removed?

File: CBLClient/Document.py [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+        args.setMemoryPointer("document", document)
+        args.setString("key", key)
+        return self._client.invokeMethod("document_contains", args)
+
+    def count(self, document):
+        args = Args()
+        args.setMemoryPointer("document", document)
+        return self._client.invokeMethod("document_count", args)
+
+#     def get(self, document, key):
```
Comment (raghusarangapani) [inline CBLClient/Document.py]: Can these commented lines be removed?

File: CBLClient/Document.py [hunk 10]
```diff
@@ -0,0 +1,253 @@
...
+        return self._client.invokeMethod("document_getLong", args)
+
+    def setLong(self, document, key, value):
+        args = Args()
+        args.setMemoryPointer("document", document)
+        args.setString("key", key)
+        args.setLong("value", value)
+        return self._client.invokeMethod("document_setLong", args)
+
+#     def getObject(self, document, key):
```
Comment (raghusarangapani) [inline CBLClient/Document.py]: Can these commented lines be removed?

File: CBLClient/Document.py [hunk 11]
```diff
@@ -0,0 +1,253 @@
...
+        args = Args()
+        args.setMemoryPointer("document", document)
+        return self._client.invokeMethod("document_toMap", args)
+
+    def toMutable(self, document):
+        args = Args()
+        args.setMemoryPointer("document", document)
+        return self._client.invokeMethod("document_toMutable", args)
+
+#     def toDictionary(self, document):
```
Comment (raghusarangapani) [inline CBLClient/Document.py]: Can these commented lines be removed?

File: CBLClient/Query.py [hunk 12]
```diff
@@ -0,0 +1,409 @@
...
+
+    def query_select_result_expression_create(self, expression):
+        args = Args()
+        args.setMemoryPointer("expression", expression)
+
+        return self._client.invokeMethod("selectResult_expressionCreate", args)
+
+    def query_select_result_all_create(self):
+        # needs to be change
+        # return self._client.invokeMethod("query_select_result_all_create")
```
Comment (raghusarangapani) [inline CBLClient/Query.py]: Can this commented line be removed?

File: CBLClient/Query.py [hunk 13]
```diff
@@ -0,0 +1,409 @@
...
+
+        return self._client.invokeMethod("query_getDoc", args)
+
+    def query_get_docs_limit_offset(self, database, limit, offset):
+        args = Args()
+        args.setMemoryPointer("database", database)
+        args.setInt("limit", limit)
+        args.setInt("offset", offset)
+
+        # needs to be change
```
Comment (raghusarangapani) [inline CBLClient/Query.py]: Can these commented lines be removed?

File: CBLClient/Query.py [hunk 14]
```diff
@@ -0,0 +1,409 @@
...
+
+    def query_multiple_selects(self, database, select_property1, select_property2, whr_key, whr_val):
+        args = Args()
+        args.setMemoryPointer("database", database)
+        args.setString("select_property1", select_property1)
+        args.setString("select_property2", select_property2)
+        args.setString("whr_key", whr_key)
+        args.setString("whr_val", whr_val)
+
+        # needs to be change
```
Comment (raghusarangapani) [inline CBLClient/Query.py]: Can these commented lines be removed?

File: CBLClient/Query.py [hunk 15]
```diff
@@ -0,0 +1,409 @@
...
+        args.setString("whr_key1", whr_key1)
+        args.setString("whr_val1", whr_val1)
+        args.setString("whr_key2", whr_key2)
+        args.setString("whr_val2", whr_val2)
+        args.setString("whr_key3", whr_key3)
+        args.setString("whr_val3", whr_val3)
+        args.setString("whr_key4", whr_key4)
+        args.setBoolean("whr_val4", whr_val4)
+
+        # return self._client.invokeMethod("query_where_and_or", args)
```
Comment (raghusarangapani) [inline CBLClient/Query.py]: Can this commented line be removed?

File: CBLClient/Replication.py [hunk 16]
```diff
@@ -0,0 +1,351 @@
...
+        if target_db is not None:
+            args.setMemoryPointer("target_db", target_db)
+
+        cluster_config = os.environ["CLUSTER_CONFIG"]
+        if sg_ssl_enabled(cluster_config):
+            args.setString("pinnedservercert", "sg_cert")
+
+        return self._client.invokeMethod("replicatorConfiguration_configure", args)
+
+    """def create(self, source_db, target_db=None, target_url=None):
```
Comment (raghusarangapani) [inline CBLClient/Replication.py]: Can these commented lines be removed?

File: CBLClient/Replication.py [hunk 17]
```diff
@@ -0,0 +1,351 @@
...
+        if target_db:
+            args.setMemoryPointer("targetDb", target_db)
+        elif target_url:
+            args.setMemoryPointer("targetURI", target_url)
+        else:
+            raise Exception("Pass either target_db or target_url.")
+        return self._client.invokeMethod("replicatorConfiguration_create",
+                                         args) """
+
+#     Protected method now
```
Comment (raghusarangapani) [inline CBLClient/Replication.py]: Can these commented lines be removed?

File: CBLClient/Replication.py [hunk 18]
```diff
@@ -0,0 +1,351 @@
...
+        args = Args()
+        args.setMemoryPointer("config", config)
+        return self._client.invokeMethod("replicator_create", args)
+
+    def getConfig(self, replicator):
+        args = Args()
+        args.setMemoryPointer("replicator", replicator)
+        return self._client.invokeMethod("replicator_config", args)
+
+#     Same as status method
```
Comment (raghusarangapani) [inline CBLClient/Replication.py]: Can these commented lines be removed?

File: CBLClient/Replication.py [hunk 19]
```diff
@@ -0,0 +1,351 @@
...
+        args.setMemoryPointer("replicator", replicator)
+        args.setMemoryPointer("changeListener", change_listener)
+        return self._client.invokeMethod("replicator_removeChangeListener", args)
+
+    def toString(self, replicator):
+        args = Args()
+        args.setMemoryPointer("replicator", replicator)
+        return self._client.invokeMethod("replicator_toString", args)
+
+#     Not available any more
```
Comment (raghusarangapani) [inline CBLClient/Replication.py]: Can these commented lines be removed?

File: CBLClient/Replication.py [hunk 20]
```diff
@@ -0,0 +1,351 @@
...
+#     def networkReachable(self, replicator):
+#         args = Args()
+#         args.setMemoryPointer("replicator", replicator)
+#         return self._client.invokeMethod("replicator_networkReachable", args)
+#
+#     def networkUnreachable(self, replicator):
```
Comment (raghusarangapani) [inline CBLClient/Replication.py]: Can these commented lines be removed?

File: CBLClient/Replication.py [hunk 21]
```diff
@@ -0,0 +1,351 @@
...
+
+    def getActivitylevel(self, replicator):
+        args = Args()
+        args.setMemoryPointer("replicator", replicator)
+        return self._client.invokeMethod("replicator_getActivityLevel", args)
+
+    def getError(self, replicator):
+        args = Args()
+        args.setMemoryPointer("replicator", replicator)
+#         error = self._client.invokeMethod("replicator_getError", args)
```
Comment (raghusarangapani) [inline CBLClient/Replication.py]: Can these commented lines be removed?

File: CBLClient/Replication.py [hunk 22]
```diff
@@ -0,0 +1,351 @@
...
+
+    def getChangesCount(self, change_listener):
+        args = Args()
+        args.setMemoryPointer("changeListener", change_listener)
+        return self._client.invokeMethod("replicator_changeListenerChangesCount", args)
+
+    def getChangesChangeListener(self, change_listener):
+        args = Args()
+        args.setMemoryPointer("changeListener", change_listener)
+        # args.setInt("index", index)
```
Comment (raghusarangapani) [inline CBLClient/Replication.py]: Can this commented line be removed?

File: CBLClient/Replication.py [hunk 23]
```diff
@@ -0,0 +1,351 @@
...
+            activity_level = self.getActivitylevel(repl)
+
+    def create_session_configure_replicate(self, baseUrl, sg_admin_url, sg_db, username, password,
+                                           channels, sg_client, cbl_db, sg_blip_url, replication_type=None, continuous=True):
+
+        authenticator = Authenticator(baseUrl)
+        cookie, session_id = sg_client.create_session(sg_admin_url, sg_db, username)
+        session = cookie, session_id
+        replicator_authenticator = authenticator.authentication(session_id, cookie, authentication_type="session")
+        # replicator_authenticator = authenticator.authentication(username=username, password=password, authentication_type="basic")
```
Comment (raghusarangapani) [inline CBLClient/Replication.py]: Can these commented lines be removed?

File: CBLClient/Replication.py [hunk 24]
```diff
@@ -0,0 +1,351 @@
...
+        repl_config = self.configure(cbl_db, sg_blip_url, continuous=continuous, channels=channels, replication_type=replication_type, replicator_authenticator=replicator_authenticator)
+        repl = self.create(repl_config)
+        self.start(repl)
+        self.wait_until_replicator_idle(repl)
+
+        return session, replicator_authenticator, repl
+
+
+"""
+    def configure(self, source_db, target_url=None, target_db=None,
```
Comment (raghusarangapani) [inline CBLClient/Replication.py]: Can these commented lines be removed?

File: CBLClient/RestClient.py [hunk 25]
```diff
@@ -0,0 +1,43 @@
+from requests import Session
```
Comment (raghusarangapani) [inline CBLClient/RestClient.py]: Why is this file needed? The APIs here are already part of other .py files

File: CBLClient/TestCase.py [hunk 26]
```diff
@@ -0,0 +1,243 @@
+from Client import Client
```
Comment (raghusarangapani) [inline CBLClient/TestCase.py]: This file is probably not needed anymore

File: CBLClient/Client.py (hunk 1 above)
Comment (hrajput89) [reply CBLClient/Client.py]: done

File: CBLClient/Database.py (hunk 2 above)
Comment (hrajput89) [reply CBLClient/Database.py]: done

File: CBLClient/DatabaseConfiguration.py (hunk 3 above)
Comment (hrajput89) [reply CBLClient/DatabaseConfiguration.py]: done

File: CBLClient/Dictionary.py (hunk 4 above)
Comment (hrajput89) [reply CBLClient/Dictionary.py]: done

File: CBLClient/Dictionary.py (hunk 5 above)
Comment (hrajput89) [reply CBLClient/Dictionary.py]: done

File: CBLClient/Dictionary.py (hunk 6 above)
Comment (hrajput89) [reply CBLClient/Dictionary.py]: done

File: CBLClient/Document.py (hunk 7 above)
Comment (hrajput89) [reply CBLClient/Document.py]: Removed

File: CBLClient/Document.py (hunk 8 above)
Comment (hrajput89) [reply CBLClient/Document.py]: done

File: CBLClient/Document.py (hunk 9 above)
Comment (hrajput89) [reply CBLClient/Document.py]: done

File: CBLClient/Document.py (hunk 10 above)
Comment (hrajput89) [reply CBLClient/Document.py]: done

File: CBLClient/Document.py (hunk 11 above)
Comment (hrajput89) [reply CBLClient/Document.py]: done

File: CBLClient/Query.py (hunk 13 above)
Comment (hrajput89) [reply CBLClient/Query.py]: done

File: CBLClient/Query.py (hunk 14 above)
Comment (hrajput89) [reply CBLClient/Query.py]: done

File: CBLClient/Replication.py (hunk 16 above)
Comment (hrajput89) [reply CBLClient/Replication.py]: done

File: CBLClient/Replication.py (hunk 17 above)
Comment (hrajput89) [reply CBLClient/Replication.py]: done

File: CBLClient/Replication.py (hunk 18 above)
Comment (hrajput89) [reply CBLClient/Replication.py]: done

File: CBLClient/Replication.py (hunk 19 above)
Comment (hrajput89) [reply CBLClient/Replication.py]: done

File: CBLClient/Replication.py (hunk 20 above)
Comment (hrajput89) [reply CBLClient/Replication.py]: done

File: CBLClient/Replication.py (hunk 21 above)
Comment (hrajput89) [reply CBLClient/Replication.py]: done

File: CBLClient/Replication.py (hunk 22 above)
Comment (hrajput89) [reply CBLClient/Replication.py]: done

File: CBLClient/Replication.py (hunk 23 above)
Comment (hrajput89) [reply CBLClient/Replication.py]: done

File: CBLClient/Replication.py (hunk 24 above)
Comment (hrajput89) [reply CBLClient/Replication.py]: done

File: CBLClient/TestCase.py (hunk 26 above)
Comment (hrajput89) [reply CBLClient/TestCase.py]: deleted

File: CBLClient/RestClient.py (hunk 25 above)
Comment (hrajput89) [reply CBLClient/RestClient.py]: deleted

**Comment by hrajput89** [comment]
Made changes as suggested by @raghusarangapani

File: CBLClient/Authenticator.py [hunk 27]
```diff
@@ -0,0 +1,68 @@
...
+        return self._client.invokeMethod("sessionAuthenticator_getSessionId",
+                                         args)
+
+    def sessionAuthenticator_getCookieName(self, session):
+        args = Args()
+        args.setMemoryPointer("session", session)
+        return self._client.invokeMethod("sessionAuthenticator_getCookieName",
+                                         args)
+
+#     def sessionAuthenticator_getExpires(self, session):
```
Comment (raghusarangapani) [inline CBLClient/Authenticator.py]: Can this commented API be removed?

File: CBLClient/Database.py [hunk 28]
```diff
@@ -0,0 +1,318 @@
...
+        return self._client.invokeMethod("database_updateDocuments", args)
+
+    def updateDocument(self, database, data, doc_id):
+        args = Args()
+        args.setMemoryPointer("database", database)
+        args.setDictionary("data", data)
+        args.setString("id", doc_id)
+        return self._client.invokeMethod("database_updateDocument", args)
+
+#     def contains(self, database, doc_id):
```
Comment (raghusarangapani) [inline CBLClient/Database.py]: Can this commented API be removed?

Comment (sridevi-15) [reply CBLClient/Database.py]: hope this is removed

File: CBLClient/ValueSerializer.py [hunk 29]
```diff
@@ -0,0 +1,103 @@
...
+        elif isinstance(value, dict):
+            dict_map = value
+            string_map = {}
+
+            for map_param in dict_map:
+                val = ValueSerializer.serialize(dict_map[map_param])
+                string_map[map_param] = val
+
+            return json.dumps(string_map)
+            # return json.dumps(value)
```
Comment (sridevi-15) [inline CBLClient/ValueSerializer.py]: you can remove this

File: CBLClient/Authenticator.py (hunk 27 above)
Comment (hrajput89) [reply CBLClient/Authenticator.py]: removed it.

File: CBLClient/ValueSerializer.py (hunk 29 above)
Comment (hrajput89) [reply CBLClient/ValueSerializer.py]: removed it.

File: CBLClient/Query.py (hunk 12 above)
Comment (raghusarangapani) [reply CBLClient/Query.py]: Seems like this comment is not removed?

File: CBLClient/Query.py (hunk 15 above)
Comment (raghusarangapani) [reply CBLClient/Query.py]: Seems like this comment is not removed
